File: ChatSupport/src/main/java/com/vantagecircle/chatapp/core/interfaceC/ChildInterface.java
package com.vantagecircle.chatapp.core.interfaceC;

import com.vantagecircle.chatapp.core.model.DataModel;

/**
 * Created by bapidas on 01/08/17.
 */

public interface ChildInterface {
    void onChildNew(DataModel dataModel);

    void onChildModified(DataModel dataModel);

    void onChildDelete(DataModel dataModel);

    void onChildRelocate(DataModel dataModel);

    void onChildCancelled(DataModel dataModel);
}

File: app/src/main/java/com/bapidas/chattingapp/data/core/callbacks/ChildInterface.kt
package com.bapidas.chattingapp.data.core.callbacks

import com.bapidas.chattingapp.data.core.model.DataModel

/**
 * Created by bapidas on 01/08/17.
 */
interface ChildInterface {
    fun onChildNew(dataModel: DataModel)
    fun onChildModified(dataModel: DataModel)
    fun onChildDelete(dataModel: DataModel)
    fun onChildRelocate(dataModel: DataModel)
    fun onChildCancelled(dataModel: DataModel)
}